test_util/cluster_api.py

```diff
@@ -338,6 +340,17 @@ def get_test_app_in_docker(self, ip_per_container):
         return app, test_uuid
 
     def deploy_test_app_and_check(self, app, test_uuid):
+        """This method deploys the test server app and then
+        pings its /operating_environment endpoint to retrieve the container
+        user.
+
+        In a docker container this will be whatever is set as USER (defaults to root)
+        In a mesos container, this will be the marathon user
+        """
+        if 'container' in app and app['container']['type'] == 'DOCKER':
+            marathon_user = 'root'
+        else:
+            marathon_user = self.default_os_user
```

Comment:
Shouldn't this do a app.get('user', self.default_os_user) so the user can be overridden via the marathon.json?